sdk/core/core-amqp/src/errors.ts

```diff
@@ -539,6 +549,29 @@ function isBrowserWebsocketError(err: any): boolean {
   return result;
 }
 
+/**
+ * @internal
+ * Checks if given object maps to a valid custom error. If yes, configures and returns the appropriate error instance, else returns `undefined`.
+ * @param err
+ */
+function getCustomError(err: AmqpError | Error): MessagingError | undefined {
+  const error: MessagingError = err as MessagingError;
+  if (
+    // instanceof checks on custom Errors doesn't work without manually setting the prototype within the error.
+    // Must do a name check until the custom error is updated, and that doesn't break compatibility
+    // https://github.com/Microsoft/TypeScript/wiki/Breaking-Changes#extending-built-ins-like-error-array-and-map-may-no-longer-work
+    retryableErrors.indexOf((err as Error).name) > 0
+  ) {
+    error.retryable = true;
+    return error;
+  } else if (nonRetryableErrors.indexOf((err as Error).name) > 0) {
```

Comment:
If a preceding `if` block has a `return` in the end, then the following `else if` can be a simpler `if`.

Reply:
Updated and addressed other comments as well